app/policies/scm/authorization_policy.rb

```diff
@@ -0,0 +1,72 @@
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+# This capsulates permissions a user has for a work package.  It caches based
+# on the work package's project and is thus optimized for the context menu.
+#
+# This is no conern but it was placed here so that it will be removed together
+# with the rest of the experimental API.
+
+class Scm::AuthoriziationPolicy
+  attr_reader :project, :user
+
+  def initialize(project, user)
+    @user = user
+    @project = project
+  end
+
+  ##
+  # Determines the authorization status for the user of this project
+  # for a given repository request.
+  # May be overridden by descendents when more than the read/write distinction
+  # is necessary.
+  def authorized?(params)
+    (readonly_request?(params) && read_access?) || write_access?
+  end
+
+  protected
+
+  ##
+  # Determines whether the given request is a read access
+  # Must be implemented by descendents of this policy.
+  def readonly_request?(params)
```

Comment:
Unused method argument - `params`. If it's necessary, use `_` or `_params` as an argument name to indicate that it won't be used. You can also write as `readonly_request?(*)` if you want the method to accept any arguments but don't care about them.